0007-onion-routing-over-libp2p.adoc

```diff
@@ -0,0 +1,126 @@
+= Improving privacy through onion routing over libp2p
+Daniel Karzel <daniel.karzel@coblox.tech>;
+:toc:
+:revdate: 2019-06-17
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/coblox/spikes/issues/8[Tor Support for COMIT]
+
+== Context
+
+Onion routing as implemented by the https://en.wikipedia.org/wiki/Tor_(anonymity_network)[Tor network] enables anonymous usage of the internet
+through end2end multi-time-encryption over multiple hops provided through relay nodes. Anybody in the network can set up a relay node in the network.
+
+Onion routing would help to create more privacy through anonymity in the comit network.
+
+
+== Research
+
+=== Onion Routing over libp2p
```

Comment:
Did you have a look at how bitcoind and lightning network does it? 
What I found:
* bitcoind allows usage of the tor network (https://en.bitcoin.it/wiki/Setting_up_a_Tor_hidden_service) 
* LN can be run in the same way (https://medium.com/@notgrubles/setup-a-private-lightning-node-using-blockstreams-c-lightning-over-tor-12773ade6d90)

Reply:
This is even more detailed: https://github.com/ElementsProject/lightning/blob/master/doc/TOR.md
Let's discuss the whole issue face2face and then I'll update the spike.